File: cloud_agents/exposed_service_agent/src/main/java/com/netteans/cloud/explosed/service/BootStrap.java
package com.netteans.cloud.explosed.service;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.SpringApplication;
import org.springframework.cloud.client.SpringCloudApplication;
import org.springframework.cloud.netflix.hystrix.EnableHystrix;
import org.springframework.cloud.openfeign.EnableFeignClients;

//@EnableDiscoveryClient
@EnableHystrix
@EnableFeignClients
@SpringCloudApplication
public class BootStrap {
    private static Logger logger = LoggerFactory.getLogger(BootStrap.class);
    private static String port;

//    @Value("${server.port}")
    private void setPort(String p) {
        port = p;
    }

    public static void main(String[] args) {
        SpringApplication.run(BootStrap.class, args);
        logger.info("port is {}", port); //启动之后才会注入
    }
}

File: cloud_agents/exposed_service_agent_kt/src/main/kotlin/com/netteans/cloud/exposed/service/kt/BootStrap.kt
package com.netteans.cloud.exposed.service.kt

import org.slf4j.LoggerFactory
import org.springframework.boot.SpringApplication
import org.springframework.cloud.client.SpringCloudApplication
import org.springframework.cloud.netflix.hystrix.EnableHystrix
import org.springframework.cloud.openfeign.EnableFeignClients

//@EnableDiscoveryClient
@EnableHystrix
@EnableFeignClients
@SpringCloudApplication
open class BootStrap {

    //    @Value("${server.port}")
    private fun setPort(p: String) {
        port = p
    }

    companion object {
        private val logger = LoggerFactory.getLogger(BootStrap::class.java)
        private var port: String? = null

        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(BootStrap::class.java, *args)
            logger.info("port is {}", port) //启动之后才会注入
        }
    }
}
